src/core/models/section.js

```diff
@@ -81,10 +83,16 @@ export default class Section {
 class AppliedQueryFilter {
   // Support legacy model and new model until fully migrated.
   // TODO(billy) Remove the left expression during assignment when migrated.
+  // TODO(SPR-2394): convert this into a FilterNode here instead of in VerticalResults
   constructor (appliedQueryFilter) {
     this.key = appliedQueryFilter.key || appliedQueryFilter.displayKey;
     this.value = appliedQueryFilter.value || appliedQueryFilter.displayValue;
-    this.fieldId = Object.keys(appliedQueryFilter.filter)[0];
+    this.filter = Filter.from(appliedQueryFilter.filter);
+    this.fieldId = this.filter.getFilterKey();
+    this.metadata = new FilterMetadata({
+      fieldName: this.key,
+      displayValue: this.value
+    });
```

Comment:
Minor nit, but I'd suggest we not mess with this class until we do SPR-2394. We can have VerticalResults do the full conversion for now.

Reply:
sure, I thought this was pretty safe and would make things 1% easier later but updated